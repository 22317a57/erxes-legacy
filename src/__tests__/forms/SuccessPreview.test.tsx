import { mount, shallow } from 'enzyme';
import SuccessPreview from 'modules/forms/components/step/preview/SuccessPreview';
import * as React from 'react';
import * as renderer from 'react-test-renderer';

describe('SuccessPreview component', () => {
  const defaultProps = {
    onChange: (name: any, value: string) => null,
    color: 'red',
    theme: 'def'
  };

  test('renders shallow successfully', () => {
    shallow(<SuccessPreview {...defaultProps} />);
  });

  test('renders mount with default props', () => {
    const control = mount(<SuccessPreview {...defaultProps} />);
    const props = control.props();

    expect(props).toMatchObject(defaultProps);
  });

  test('snapshot matches', () => {
    const rendered = renderer
      .create(<SuccessPreview {...defaultProps} />)
      .toJSON();

    expect(rendered).toMatchSnapshot();
  });
});
