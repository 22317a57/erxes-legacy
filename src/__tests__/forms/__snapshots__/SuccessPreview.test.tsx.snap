// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SuccessPreview component snapshot matches 1`] = `
<div
  className="sc-aewfc dsStAJ"
>
  <div
    className="sc-isBZXS htStnM"
  >
    <div
      className="sc-OxbzP jocSpk"
      style={
        Object {
          "backgroundColor": "def",
        }
      }
    >
      <div />
    </div>
    <div
      className="sc-lnrBVv jitmGX"
    >
      <div
        className="sc-fvLVrH kjchyg"
      >
        <div
          className="sc-dBAPYN igvoAR"
        />
      </div>
    </div>
  </div>
</div>
`;
